2021_RTCSA_dynfed/src/main.rs

```diff
@@ -31,31 +37,83 @@ struct ArgParser {
     output_dir_path: String,
 }
 
+pub fn adjust_dag_set(dag_set: &mut [Graph<NodeData, i32>]) {
+    for dag in dag_set.iter_mut() {
+        adjust_dag(dag);
+    }
+}
+
+fn adjust_dag(dag: &mut Graph<NodeData, i32>) {
+    if let (Some(period), Some(end_to_end_deadline)) =
+        (dag.get_head_period(), dag.get_end_to_end_deadline())
+    {
+        handle_equal_period_and_deadline(dag, period, end_to_end_deadline);
+    } else if dag.get_head_period().is_none() {
```

Comment:
It is recommended to store the result in a temporary variable, as it is inefficient to call `get_head_period` and `get_end_to_end_deadline` for every conditional statement.

Reply:
Refactoring has resulted in the use of match statements.